File: dofun/safe/deviceinfo/utils/DebugUtils.java
package com.dofun.safe.deviceinfo.utils;

import android.content.Context;
import android.os.Debug;
import android.provider.Settings;

/* loaded from: classes6.dex */
public class DebugUtils {
    public static boolean isOpenDebug(Context context) {
        try {
            return Settings.Secure.getInt(context.getContentResolver(), "adb_enabled", 0) > 0;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    public static boolean isDebugVersion(Context context) {
        try {
            return (context.getApplicationInfo().flags & 2) != 0;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    public static boolean isDebugConnected() {
        try {
            return Debug.isDebuggerConnected();
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    public static String getUsbDebugStatus() {
        return CommandUtils.execute("getprop init.svc.adbd");
    }
}

File: app/src/main/java/com/dofun/safe/deviceinfo/utils/DebugUtils.kt
package com.dofun.safe.deviceinfo.utils

import android.content.Context
import android.os.Debug
import android.provider.Settings

object DebugUtils {
    fun isOpenDebug(context: Context): Boolean {
        return try {
            Settings.Secure.getInt(context.contentResolver, "adb_enabled", 0) > 0
        } catch (e: Exception) {
            e.printStackTrace()
            false
        }
    }

    fun isDebugVersion(context: Context): Boolean {
        return try {
            context.applicationInfo.flags and 2 != 0
        } catch (e: Exception) {
            e.printStackTrace()
            false
        }
    }

    fun isDebugConnected(): Boolean {
        return try {
            Debug.isDebuggerConnected()
        } catch (e: Exception) {
            e.printStackTrace()
            false
        }
    }

    fun getUsbDebugStatus(): String? {
        return CommandUtils.execute("getprop init.svc.adbd")
    }
}
